sandy/fy.py

```diff
@@ -744,40 +1166,41 @@ def gls_update(self, zam, e, y_extra=None, Vy_extra=None,
         mask = (data.ZAM == zam) & (data.MT == 454) & (data.E == e)
         data = data.loc[~mask]
         x_prior = fy_data.query('MT==454').FY
-        index = x_prior.index
         Vx_prior = fy_data.query('MT==454').DFY
+        index = x_prior.index
         Vx_prior = sandy.CategoryCov.from_var(Vx_prior).data
         # Find the GLS varibles:
         if kind == 'mass yield':
             model_sensitivity_object = self._filters(conditions)
         elif kind == 'cumulative' or 'chain yield':
             model_sensitivity_object = decay_data
-        S = _gls_setup(model_sensitivity_object, kind)
+        S = _gls_setup(model_sensitivity_object, kind, **kwargs)
         # Perform GLS:
         if y_extra is None and Vy_extra is None:
             x_post = sandy.constrained_gls_update(x_prior, S, Vx_prior,
+                                                  rows=rows,
                                                   threshold=threshold)
             Vx_post = self.gls_cov_update(zam, e, kind=kind,
-                                          decay_data=decay_data,
-                                          threshold=threshold)
+                                          decay_data=decay_data, rows=rows,
+                                          threshold=threshold).data
         else:
             x_post = sandy.gls_update(x_prior, S, Vx_prior, Vy_extra, y_extra,
```

Comment:
Therefore, in that implementation you would only be updating one number, since y_calc = S.dot(theta), and if S and theta are vectors, y_calc is a scalar. Therefore, we would only update one number and its associated error. Is this an interesting application?

Reply:
Wait: As I said, I mean in the GLS technique not sandwitch formula. It could be useful to insert a further constain for the fission yield such as the sum of the fission yields equals to 2 (which means a sensitivity vector equals to a unit vector). In the formula `V_{x_{post}} = V_{x_{prior}} - V_{x_{prior}}\cdot S.T \cdot \left(S\cdot V_{x_{prior}}\cdot S.T + V_{y_{extra}}\right)^{-1} \cdot S \cdot V_{x_{prior}}` even if S is a vector you will obtain `V_{x_{post}}` which is a matrix, so sure! It is an intresting application I think!

Reply:
If you think that S as a vector is interesting, I'll implement it in another branch, since the whole GLS update is done for a matrix.

Reply:
On the other hand, we should not introduce constrains. It is the data that must carry the constrains itself. That is, even if we know that fy has to be two, we do not have enough data to apply a correct weight function and normalise the results. The failure comes from the model we are using or from the original data (mostly these).

Reply:
Maybe I didn't explain myself properly. Did you see the update scheme in https://doi.org/10.1016/j.anucene.2015.10.027? 4 of the 5 update steps are performed with S as a vector. I hope I made my point more clear :)